src/classes/data1dstore.cpp

```diff
@@ -18,47 +14,32 @@ bool Data1DStore::addData(std::string_view dataName, LineParser &parser, int sta
                           const CoreData &coreData)
 {
     // Create new data
-    Data1D *data = data_.add();
-    data->setName(dataName);
-
-    // Add reference to data
-    RefDataItem<Data1D, Data1DImportFileFormat> *ref = dataReferences_.append(data);
+    auto &data = data_.emplace_back();
+    data.first.setTag(dataName);
```

Comment:
```suggestion
    auto &[data, format] = data_.emplace_back();
    data.setTag(dataName);
```
Use structured binding